namui/src/namui/namui_web/fetch.rs

```diff
@@ -1,50 +1,69 @@
+use core::fmt;
+
 use js_sys::{ArrayBuffer, Uint8Array};
-use wasm_bindgen::JsCast;
+use wasm_bindgen::{JsCast, JsValue};
 use wasm_bindgen_futures::JsFuture;
 use web_sys::{Request, RequestInit, Response};
 
-pub async fn fetch_get(url: &str) -> Result<Response, String> {
+pub async fn fetch_get(url: &str) -> Result<Response, FetchError> {
     let mut options = RequestInit::new();
     options.method("GET");
 
-    let request = Request::new_with_str_and_init(url, &options).unwrap();
+    let request = Request::new_with_str_and_init(url, &options)?;
 
     let window = web_sys::window().unwrap();
-    let response_value = JsFuture::from(window.fetch_with_request(&request))
-        .await
-        .unwrap();
-    assert!(response_value.is_instance_of::<Response>());
-    let response: Response = response_value.dyn_into().unwrap();
+    let response_value = JsFuture::from(window.fetch_with_request(&request)).await?;
+
+    let response: Response = response_value.dyn_into()?;
 
     if !response.ok() {
-        return Err(response.status_text());
+        return Err(FetchError {
+            message: response.status_text(),
+        });
     }
     Result::Ok(response)
 }
 
-pub async fn fetch_get_array_buffer(url: &str) -> Result<ArrayBuffer, String> {
-    let response: Response = fetch_get(&url).await.unwrap();
+pub async fn fetch_get_array_buffer(url: &str) -> Result<ArrayBuffer, FetchError> {
+    let response: Response = fetch_get(&url).await?;
 
-    let array_buffer = JsFuture::from(response.array_buffer().unwrap())
-        .await
-        .unwrap()
-        .dyn_into()
-        .unwrap();
+    let array_buffer = JsFuture::from(response.array_buffer()?).await?.dyn_into()?;
 
     Result::Ok(array_buffer)
 }
 
-pub async fn fetch_get_json<T: for<'a> serde::Deserialize<'a>>(url: &str) -> Result<T, String> {
-    let response: Response = fetch_get(&url).await.unwrap();
+pub async fn fetch_get_json<T: for<'a> serde::Deserialize<'a>>(url: &str) -> Result<T, FetchError> {
+    let response: Response = fetch_get(&url).await?;
 
-    let json = JsFuture::from(response.json().unwrap()).await.unwrap();
+    let json = JsFuture::from(response.json()?).await?;
 
-    json.into_serde().map_err(|e| e.to_string())
+    json.into_serde().map_err(|e| FetchError {
+        message: format!("fail to deserialize for {} - {}", url, e),
+    })
 }
 
-pub async fn fetch_get_vec_u8(url: &str) -> Result<Vec<u8>, String> {
-    let array_buffer = fetch_get_array_buffer(url).await.unwrap();
+pub async fn fetch_get_vec_u8(url: &str) -> Result<Vec<u8>, FetchError> {
+    let array_buffer = fetch_get_array_buffer(url).await?;
     let array_buffer_view = Uint8Array::new(&array_buffer);
     let bytes = array_buffer_view.to_vec();
     Result::Ok(bytes)
 }
+
+#[derive(Debug)]
+pub struct FetchError {
```

Comment:
Most of Err type in this code is JsValue. so it's not possible to implement `From<String> for String` because that struct is not defined by our crate. 
So I created new struct and wrap errors with it.